api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.openmrs.Condition;
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.ConditionDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This class implements the {@link ConditionService} interface
+ * It defines the methods to handle the condition domain object
+ *
+ * @since 2.2
+ */
+@Transactional
+public class ConditionServiceImpl extends BaseOpenmrsService implements ConditionService{
+
+	private ConditionDAO conditionDAO;
+
+	public void setConditionDAO(ConditionDAO conditionDAO) {
+		this.conditionDAO = conditionDAO;
+	}
+
+	public ConditionDAO getConditionDAO() {
+		return conditionDAO;
+	}
+
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 */
+	@Override
+	public Condition saveCondition(Condition condition) throws APIException {
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Gets a condition based on the uuid
+	 *
+	 * @param uuid - uuid of the condition to be returned
+	 * @return the condition
+	 */
+	@Override
+	@Transactional(readOnly = true)
```

Comment:
Yu're doing the apposite, class level should be readOnly, and then on methods like save, void, unvoid and purge  you can override it to false

Reply:
I have reverted this

Reply:
Based on the existing convention as can be seen in other service implementations, @Transaction is used at the class level. Then for methods that do not need write access like the get methods, we define @Transaction(readOnly = true). This what I implemented as @dkayiwa called my attention to it.